pkg/controller/v1beta1/inferenceservice/reconcilers/keda/keda_reconciler.go

```
@@ -0,0 +1,212 @@
+}
+
+func NewKedaReconciler(client client.Client,
+	clientset kubernetes.Interface,
+	scheme *runtime.Scheme,
+	componentMeta metav1.ObjectMeta,
+	componentExt *v1beta1.ComponentExtensionSpec) *KedaReconciler {
+	return &KedaReconciler{
+		client:       client,
+		scheme:       scheme,
+		ScaledObject: createKedaScaledObject(clientset, componentMeta, componentExt),
+		componentExt: componentExt,
+	}
+}
+
+func getKedaMetrics(metricConfig *v1beta1.MetricsConfig, metadata metav1.ObjectMeta,
+	componentExt *v1beta1.ComponentExtensionSpec) []kedav1alpha1.ScaleTriggers {
+	var triggers []kedav1alpha1.ScaleTriggers
+	var serverAddress string
+
+	// Default values
+	triggerType := string(corev1.ResourceCPU)
+	metricType := autoscalingv2.UtilizationMetricType
+	scaleTarget := int(constants.DefaultCPUUtilization)
+
+	// Get metric configuration from metricConfig (configmap)
+	if metricConfig != nil && metricConfig.MetricBackend != "" {
+		triggerType = metricConfig.MetricBackend
+		serverAddress = metricConfig.ServerAddress
+	}
+	// override metric configuration from componentExtension if it is set
+	if componentExt.ScaleMetric != nil {
+		triggerType = string(*componentExt.ScaleMetric)
+	}
+	if componentExt.ScaleMetricType != nil {
+		metricType = *componentExt.ScaleMetricType
+	}
+	if componentExt.ScaleTarget != nil {
+		scaleTarget = *componentExt.ScaleTarget
+	}
+	// override metric configuration from componentExtension.ScalerSpec if it is set
+	if componentExt.ScalerSpec != nil {
+		if componentExt.ScalerSpec.ScaleMetric != nil {
+			triggerType = string(*componentExt.ScalerSpec.ScaleMetric)
+		}
+		if componentExt.ScalerSpec.ServerAddress != "" {
+			serverAddress = componentExt.ScalerSpec.ServerAddress
+		}
+		if componentExt.ScalerSpec.ScaleMetricType != nil {
+			scaleTarget = *componentExt.ScalerSpec.ScaleTarget
+		}
+		if componentExt.ScalerSpec.ScaleMetricType != nil {
+			metricType = *componentExt.ScalerSpec.ScaleMetricType
```

```
	// Type of metric to use. Options are Utilization, or AverageValue.
	// +optional
	ScaleMetricType *v2.MetricTargetType `json:"scaleMetricType,omitempty"`
```
Afaik you cannot have CPU with `AverageValueMetricType` should we validate early and fail fast?